test/util/util.go

```diff
@@ -436,3 +439,27 @@ func SetAdmissionCheckActive(ctx context.Context, k8sClient client.Client, admis
 		return k8sClient.Status().Update(ctx, &updatedAc)
 	}, Timeout, Interval).Should(gomega.Succeed())
 }
+
+func AwaitAndVerifyWorkloadQueueName(ctx context.Context, client client.Client, createdWorkload *kueue.Workload, wlLookupKey types.NamespacedName, jobQueueName string) {
+	gomega.Eventually(func() bool {
+		if err := client.Get(ctx, wlLookupKey, createdWorkload); err != nil {
+			return false
+		}
+		return createdWorkload.Spec.QueueName == jobQueueName
+	}, Timeout, Interval).Should(gomega.BeTrue())
```

Comment:
```suggestion
		return createdWorkload.Spec.QueueName
	}, Timeout, Interval).Should(gomega.BeEqual(jobQueueName))
```

Reply:
It asserts no error and jobQueueName

Reply:
Oh right. However, this has the problem that if the result is false, we don't know which of the 2 conditions happened. But we can simplify:

The `Eventually` should be  just for `client.Get`. You can check the queueName outside of `Eventually`, as it's not supposed to change after the Workload is created.

Reply:
I think we need Eventually here, because it is also used after "Update", so Workload is already created but not updated yet and we need to check queue name few times. And I don't think we should split the function to two "Eventually".

Reply:
Another approach would be like this: https://github.com/kubernetes-sigs/kueue/blob/9a4c73d1dbbf795e2afd0b29178b95507fc5d98b/test/e2e/e2e_test.go#L239-L243

But this is on the nit side, so let's merge.